src/diffusion_for_multi_scale_molecular_dynamics/noisers/lattice_noiser.py

```diff
@@ -1,23 +1,101 @@
+from dataclasses import dataclass
+from typing import Tuple
+
 import torch
 
 
+@dataclass(kw_only=True)
+class LatticeDataParameters:
+    """Information on the mean of the lattice parameters used for noising a sample.
+
+    TODO: this might belong elsewhere
+    """
+
+    inverse_average_density: float  # inverse of the average volume of unit cell scales by the number of atoms
+    spatial_dimension: int = 3
+
+
 class LatticeNoiser:
     """Lattice noiser.
 
     This class provides methods to generate noisy lattices.
     TODO this is a placeholder
     """
 
+    def __init__(self, lattice_parameters: LatticeDataParameters):
+        self.inverse_density = lattice_parameters.inverse_average_density
+        self.spatial_dimension = lattice_parameters.spatial_dimension
+
     @staticmethod
-    def get_noisy_lattice_vectors(real_lattice_vectors: torch.Tensor) -> torch.Tensor:
-        """Get noisy lattice vectors.
+    def _get_gaussian_noise(shape: Tuple[int]) -> torch.Tensor:
+        """Get Gaussian noise.
 
-        TODO this is a placeholder
+        Get a sample from N(0, 1) of dimensions shape.
 
         Args:
-            real_lattice_vectors: lattice vectors from the sampled data
+            shape : the shape of the sample.
 
         Returns:
-            real_lattice_vectors: a sample of noised lattice vectors. Placeholder for now.
+            gaussian_noise: a sample from N(0, 1) of dimensions shape.
         """
-        return real_lattice_vectors
+        return torch.randn(shape)
+
+    def get_noisy_lattice_vectors(
+        self,
+        real_lattice_parameters: torch.Tensor,
+        sigmas_n: torch.Tensor,
+        alpha_bars: torch.Tensor,
+        num_atoms: torch.Tensor,
+    ) -> torch.Tensor:
+        r"""Get noisy lattice vectors.
+
+        We consider the lattice parameters as a tensor of spatial_dimension degrees of freedom i.e. we assume the unit
+        cell is orthogonal.
+
+        .. math::
+
+            q(L_t | L_0) = \mathcal{N}\left(\sqrt{\bar{\alpha}_t}L_0 +
+                (1 - \sqrt{\bar{\alpha}_t})\mu(n)I, (1-\sqrt{\bar{\alpha}_t})\sigma^2_t(n)I\right)
+
+        Args:
+            real_lattice_parameters: lattice parameters from the sampled data. These parameters are not the lattice
+                vector, but an array of dimension [spatial_dimension] containing the size of the orthogonal box.
+                TODO add angles
+            sigmas_n: variance of the perturbation kernel rescaled by the number of atoms. Tensor is assumed to be of
+                the same shape as real_lattice_parameters.
+            alpha_bars: cumulative noise scale. Tensor is assumed to be of the same shape as real_lattice_parameters.
+            num_atoms: number of atoms in each sample. Tensor should be a 1D tensor matching the first dimension of the
```

Comment:
The code below suggests that `num_atoms` should be at least two dimensional. Maybe there's a batch dimension here too?

Reply:
It's broadcasted to be batch, spatial_dimension * (spatial_dimension + 1) / 2. I fixed the doc string